python/mpt/tests/src/mpt/test_utils.py

```diff
@@ -0,0 +1,65 @@
+from ethereum.cancun.trie import (
+    BranchNode,
+    ExtensionNode,
+    LeafNode,
+    bytes_to_nibble_list,
+    encode_internal_node,
+)
+from ethereum_types.bytes import Bytes
+from hypothesis import given
+from hypothesis import strategies as st
+
+from cairo_addons.testing.errors import strict_raises
+from mpt.utils import check_branch_node, nibble_list_to_bytes
+
+
+def invalid_branch_node_strategy():
+    return st.builds(
+        BranchNode,
+        subnodes=st.one_of(
+            st.integers(min_value=0, max_value=2**256 - 1).map(
+                lambda x: x.to_bytes(32, "little")
+            ),
+            st.from_type(LeafNode).map(lambda x: encode_internal_node(x)),
+            st.from_type(ExtensionNode).map(lambda x: encode_internal_node(x)),
+            st.none(),
+            st.just(list()),
+        ).map(
+            lambda subnode: (
+                list([b""] * 13 + [[]] + [[]] + [subnode])
+                if isinstance(subnode, bytes)
+                else list([[]] * 13 + [b""] + [b""] + [subnode])
+            )
+        ),
+        value=st.just(b""),
+    )
```

Comment:
```suggestion
def invalid_branch_node_strategy():
    return st.builds(
        BranchNode,
        subnodes=st.lists(
            st.one_of(st.just([b""]), st.just([[]])),
            min_size=16,
            max_size=16,
            unique=False
        ),
        value=st.just(b""),
    )
```

Is this not what we want eventually?

Reply:
but if all of the subnodes are zero length, then the assertions will fail when we assert x len != 0?